rpm/Makefile

```diff
@@ -30,11 +36,21 @@ build: source vendor vendor-rpms
 	  yum install -y vendor-$(DISTRO)/$(ARCH)/yajl-1*.rpm; \
 	  yum install -y vendor-$(DISTRO)/$(ARCH)/yajl-devel*.rpm; \
 	fi
-	rpmbuild --define "_source_filedigest_algorithm md5" --define "_sourcedir $$(pwd)" --define "_srcrpmdir $$(pwd)" --define "collectd_version $(VERSION)" --define "package_version $(PKG_VERSION)" --define "build_num $(PKG_BUILD)" -ba stackdriver-agent.spec
-	mkdir result
-	cp /root/rpmbuild/RPMS/$(ARCH)/stackdriver-agent-*.rpm result/
+	rpmbuild --define "_topdir $(TOPDIR)" --define "_source_filedigest_algorithm md5" --define "_sourcedir $$(pwd)" --define "_srcrpmdir $$(pwd)" --define "collectd_version $(VERSION)" --define "package_version $(PKG_VERSION)" --define "build_num $(PKG_BUILD)" -ba stackdriver-agent.spec
+	mkdir -p $(OUTPUT_DIR)
+	cp $(TOPDIR)/RPMS/$(ARCH)/stackdriver-agent-*.rpm $(OUTPUT_DIR)/
+
+ARCH ?= `uname -m`
 
 OUTPUT_DIR ?= result
+
+ifeq ($(filter %.,$(shell realpath --relative-to . $(OUTPUT_DIR))),)  # Avoid trying to rm . or ..
+DIR_TO_RM := $(OUTPUT_DIR)
+else
+DIR_TO_RM :=
+endif
+
+ifeq ($(filter sles%,$(DISTRO)),)
```

Comment:
Similar confusion here: Do we mean to use `ifneq` here instead of `ifeq`? Is this why you were saying RPM spec is painful? :P

Reply:
Addressed above. This is a Makefile, which has its own pain points. The RPM specs are painful in a whole different way. :-)